bump.lua

```diff
@@ -696,19 +742,23 @@ function World:move(item, goalX, goalY, filter)
 end
 
 function World:check(item, goalX, goalY, filter)
+  local x,y,w,h = self:getRect(item)
+  return self:projectMove(item, x, y, w, h, goalX,goalY, filter)
+end
+
+function World:projectMove(item, x, y, w, h, goalX, goalY, filter)
   filter = filter or defaultFilter
 
```

Comment:
Okay, so here's the big `alreadyVisited` change I forecasted a couple times earlier. Really, the behavior here hasn't changed, but the implementation has.

**Before:** `filter` is wrapped with another function which keeps track of the "visited" items during collision response resolution. This creates a single use function every single time World:check() is invoked, a big source of garbage!

**Now:** Instead of wrapping `filter`, we just pass a `visited` table into the collision response functions, it still keeps track of "visited" items the same way, but without the waste.